packages/app-store/routing-forms/trpc/_router.ts

```diff
@@ -0,0 +1,100 @@
+  response?: typeof import("./response.handler").responseHandler;
+  forms?: typeof import("./forms.handler").formsHandler;
+  formQuery?: typeof import("./formQuery.handler").formQueryHandler;
+  formMutation?: typeof import("./formMutation.handler").formMutationHandler;
+  deleteForm?: typeof import("./deleteForm.handler").deleteFormHandler;
+  report?: typeof import("./report.handler").reportHandler;
+}
+
+const UNSTABLE_HANDLER_CACHE: AppRoutingFormsRouteshandlerCache = {};
+
+const appRoutingForms = router({
+  public: router({
+    response: publicProcedure.input(ZResponseInputSchema).mutation(async ({ ctx, input }) => {
+      if (!UNSTABLE_HANDLER_CACHE.response) {
+        UNSTABLE_HANDLER_CACHE.response = (await import("./response.handler")).responseHandler;
+      }
+
+      // Unreachable code but required for type safety
+      if (!UNSTABLE_HANDLER_CACHE.response) {
+        throw new Error("Failed to load handler");
+      }
+
+      return UNSTABLE_HANDLER_CACHE.response({ ctx, input });
+    }),
+  }),
+  forms: authedProcedure.query(async ({ ctx }) => {
+    if (!UNSTABLE_HANDLER_CACHE.forms) {
+      UNSTABLE_HANDLER_CACHE.forms = (await import("./forms.handler")).formsHandler;
+    }
+
+    // Unreachable code but required for type safety
+    if (!UNSTABLE_HANDLER_CACHE.forms) {
+      throw new Error("Failed to load handler");
+    }
+
+    return UNSTABLE_HANDLER_CACHE.forms({ ctx });
+  }),
+  formQuery: authedProcedure.input(ZFormQueryInputSchema).query(async ({ ctx, input }) => {
+    if (!UNSTABLE_HANDLER_CACHE.formQuery) {
+      UNSTABLE_HANDLER_CACHE.formQuery = (await import("./formQuery.handler")).formQueryHandler;
+    }
+
+    // Unreachable code but required for type safety
+    if (!UNSTABLE_HANDLER_CACHE.formQuery) {
+      throw new Error("Failed to load handler");
+    }
+
+    return UNSTABLE_HANDLER_CACHE.formQuery({ ctx, input });
+  }),
+  formMutation: authedProcedure.input(ZFormMutationInputSchema).mutation(async ({ ctx, input }) => {
+    if (!UNSTABLE_HANDLER_CACHE.formMutation) {
+      UNSTABLE_HANDLER_CACHE.formMutation = (await import("./formMutation.handler")).formMutationHandler;
+    }
+
+    // Unreachable code but required for type safety
+    if (!UNSTABLE_HANDLER_CACHE.formMutation) {
+      throw new Error("Failed to load handler");
+    }
+
+    return UNSTABLE_HANDLER_CACHE.formMutation({ ctx, input });
+  }),
+  deleteForm: authedProcedure.input(ZDeleteFormInputSchema).mutation(async ({ ctx, input }) => {
+    if (!UNSTABLE_HANDLER_CACHE.deleteForm) {
+      UNSTABLE_HANDLER_CACHE.deleteForm = (await import("./deleteForm.handler")).deleteFormHandler;
+    }
+
+    // Unreachable code but required for type safety
+    if (!UNSTABLE_HANDLER_CACHE.deleteForm) {
+      throw new Error("Failed to load handler");
+    }
+
+    return UNSTABLE_HANDLER_CACHE.deleteForm({ ctx, input });
+  }),
+
+  report: authedProcedure.input(ZReportInputSchema).query(async ({ ctx, input }) => {
```

Comment:
What do you mean?

Reply:
I mean to avoid repeating cache reading and setting logic. I think something like following could work

```javascript
const cachedResponse = (key, importer)=>{
    if (!UNSTABLE_HANDLER_CACHE[key]) {
       UNSTABLE_HANDLER_CACHE[key] = importer();
    }
    if (!UNSTABLE_HANDLER_CACHE[key]) {
       throw new Error("Failed to load handler");
    }
    return UNSTABLE_HANDLER_CACHE[key]({ ctx, input });
}

const appRoutingForms = {
   response: publicProcedure.input(ZResponseInputSchema).mutation(async ({ ctx, input }) => {
       return cachedImport('response', ()=>await import("./response.handler")).responseHandler)
   })
}
```

Reply:
Ohh yes good idea @hariombalhara, it could work, but I think we can do it in another pr.